packages/firmware/build.yaml

```diff
@@ -7,18 +7,34 @@ image: opensuse/leap:15.3
 {{end}}
 
 prelude:
-# - zypper addrepo -G https://download.opensuse.org/repositories/devel:/ARM:/Factory:/Contrib:/OdroidC2/standard/devel:ARM:Factory:Contrib:OdroidC2.repo
-# - zypper --gpg-auto-import-keys ref
 - zypper in -y wget
-#{   { template "distro_install" (dict "Values" .Values "Packages" "wget")}}
 
 steps:
-# - mkdir -p /data/boot
-# - zypper --installroot /data  in -yf odroidc2-firmware.aarch64
-# Without zypper:
-- wget https://download.opensuse.org/repositories/devel:/ARM:/Factory:/Contrib:/OdroidC2/standard/aarch64/odroidc2-firmware-{{.Values.version}}.aarch64.rpm
+{{if eq .Values.name "odroid-c2"}}
 - mkdir -p /data/boot
+- wget https://download.opensuse.org/repositories/devel:/ARM:/Factory:/Contrib:/OdroidC2/standard/aarch64/odroidc2-firmware-{{.Values.version}}.aarch64.rpm
 - rpm2cpio odroidc2-firmware-{{.Values.version}}.aarch64.rpm | cpio -idmv
+{{end}}
+
+{{if or (eq .Values.name "u-boot-rpi3") (eq .Values.name "u-boot-rpi4")}}
+- mkdir -p /data/{boot,usr}
+- wget https://download.opensuse.org/repositories/hardware:/boot/openSUSE_Factory_ARM/aarch64/{{.Values.name}}-{{.Values.version}}.aarch64.rpm
```

Comment:
As a note, in the long run we should do a checksum at least. We can't rely on zypper for arch specific packages as zypper filters for the host arch. An alternative would be to use zypper, but only when building under arm.

Reply:
We should ask @bzeller if `zypper download` could get a `--arch` flag :wink:

Reply:
Otherwise the above `wget` will fail with every rebuild of the package (increasing the release number).

Reply:
that's something we could track with our bot.. but if there is a way to avoid that I'm happy with